### PR TITLE
Adal exception factory hooked into module init

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Exceptions/AdalExceptionService.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+using System;
+using System.Globalization;
+using System.Text;
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    /// <summary>
+    /// Implementation of the <see cref="CoreExceptionService"/> that throws <see cref="AdalException"/>
+    /// </summary>
+    /// <remarks>Does not currently throw <see cref="AdalSilentTokenAcquisitionException"/> and
```
Comment (jmprieur) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Exceptions/AdalExceptionService.cs:38]: What does this remark mean? Can you please elaborate (I'm confused if this is a good thing, a bad thing, more work is required ?)

**Review by jmprieur — COMMENTED** [review]
added a question

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Exceptions/AdalExceptionService.cs (hunk 1 above)
Comment (henrik-me) [reply adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Exceptions/AdalExceptionService.cs:38]: from related PR, this is the placeholder for ADAL exceptions ensuring we follow the same path for both libraries. I assume it means those exceptions are still thrown in place and not using this structure.

---
In reply to: [205396669](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1148#discussion_r205396669) [](ancestors = 205396669)

Comment (bgavrilMS) [reply adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Exceptions/AdalExceptionService.cs:38]: Yes, exactly, I am not sure we'll ever throw those exceptions from core. Both adal and msal still throw exceptions using `throw new AdalServiceException` and do not use the factories.

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Exceptions/AdalExceptionService.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+using System.Text;
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    /// <summary>
+    /// Implementation of the <see cref="CoreExceptionService"/> that throws <see cref="AdalException"/>
+    /// </summary>
+    /// <remarks>Does not currently throw <see cref="AdalSilentTokenAcquisitionException"/> and 
+    /// <see cref="AdalUserMismatchException"/></remarks>
+    internal class AdalExceptionService : CoreExceptionService
```
Comment (henrik-me) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Exceptions/AdalExceptionService.cs:40]: >AdalExceptionService [](start = 19, length = 20)

I think I understand the name Factory used previously as this causes a bit of a name confusion with AdalServiceException...

File: msal/src/Microsoft.Identity.Client/Exceptions/MsalExceptionService.cs [hunk 3]
```diff
@@ -36,7 +36,7 @@ namespace Microsoft.Identity.Client
     /// <summary>
     /// Implementation of the <see cref="CoreExceptionService"/> that throws <see cref="MsalException"/>
     /// </summary>
-    internal class MsalExceptionFactory : CoreExceptionService
+    internal class MsalExceptionService : CoreExceptionService
```
Comment (henrik-me) [inline msal/src/Microsoft.Identity.Client/Exceptions/MsalExceptionService.cs:39]: >MsalExceptionService [](start = 19, length = 20)

this naming is confusing... sorry, perhaps going back to factory is better...

**Review by henrik-me — APPROVED** [review]
:shipit:

File: msal/src/Microsoft.Identity.Client/Exceptions/MsalExceptionService.cs (hunk 3 above)
Comment (bgavrilMS) [reply msal/src/Microsoft.Identity.Client/Exceptions/MsalExceptionService.cs:39]: I'll revert to the old name (*ExceptionFactory) and get this checked in

**Pull request closed without merging**